### PR TITLE
Add "already exists" output to try_create methods for named objects (events, mutex, etc.)

### LINKED ISSUE #229: unique_mutex::try_create should return the "already exists" state back
`unique_event` for named events does this, `unique_mutex` should too:

https://github.com/microsoft/wil/blob/fec64f9d9b2279f8352f29ff0951666549d36bec/include/wil/resource.h#L2670

See also https://github.com/microsoft/WindowsAppSDK/pull/2245 where this is being worked around

(cc @jlaanstra )

### PR DESCRIPTION
**Pull request by jlaanstra**
Fixes #229

### CODE REVIEW

**Review by jonwis — APPROVED** [review]
Looking pretty good, please add a test into wiTest.cpp as well.

**Review by jonwis — APPROVED** [review]

**Comment by jlaanstra** [comment]
Not sure why the build is unhappy. I don't have a CMake setup and there is no VS solution to do anything with.

**Comment by ChrisGuzak** [comment]
update the title/description to explain why this is useful, "to enable using named objects existence for single instance checks".

note, I helped someone on my team use these types to implement this just yesterday. the success + GetLastError() signal is confusing, this will mitigate that problem effectively.

**Review by raghavthind2005 — APPROVED** [review]

File: include/wil/resource.h [hunk 1]
```diff
@@ -3018,21 +3022,23 @@ namespace wil
...
                 return false;
             }
+            assign_to_opt_param(pAlreadyExists, (::GetLastError() == ERROR_ALREADY_EXISTS));
             storage_t::reset(handle);
             return true;
         }
 
         // Returns HRESULT for unique_semaphore_nothrow, void with exceptions for shared_event and unique_event
-        result create(LONG lInitialCount, LONG lMaximumCount, _In_opt_ PCWSTR name = nullptr, DWORD desiredAccess = SEMAPHORE_ALL_ACCESS, _In_opt_ PSECURITY_ATTRIBUTES pSemaphoreAttributes = nullptr)
+        result create(LONG lInitialCount, LONG lMaximumCount, _In_opt_ PCWSTR name = nullptr, DWORD desiredAccess = SEMAPHORE_ALL_ACCESS, _In_opt_ PSECURITY_ATTRIBUTES pSemaphoreAttributes = nullptr, _Out_opt_ bool *pAlreadyExists = nullptr)
```
Comment (ChrisGuzak) [inline include/wil/resource.h:3039]: another `pAlreadyExists`. change to `alreadyExists`

**Review by ChrisGuzak — CHANGES_REQUESTED** [review]
one more rename

**Comment by jonwis** [comment]
> Not sure why the build is unhappy. I don't have a CMake setup and there is no VS solution to do anything with.

The building/testing topic in the README shows how to get started.  Worked for me with LLVM 14, VS 2022, `.\scripts\init_all.cmd` and `.\scripts\build_all.cmd`

**Review by ChrisGuzak — APPROVED** [review]

**Pull request merged**